common/web/gesture-recognizer/src/pathSegmenter.ts

```diff
@@ -0,0 +1,957 @@
+          firstChopPoint = this.lingeringSubsegmentations[0].baseChop;
+          console.log("Will merge in old subsegments!");
+        }
+      }
+
+      // Step 2:  okay, predecessor handling complete.  Now, do we think the newly-starting right-hand
+      // subsegment is likely to be 'linked' to the newly-completed left-hand subsegment?
+
+      console.log("Double-checking right-side split subsegment for xy/yx correlation with prior segment candidate(s)");
+      const fullUnion = subsegmentation.endpoint.deaccumulate(firstChopPoint);
+      const fullMergeSegmentation = new Segmentation(predecessor, subsegmentation.post, fullUnion, subsegmentation.endpoint);
+      console.log("segmentation-prevention check:")
+      console.log(fullMergeSegmentation);
+      // if(!force && PathSegmenter.shouldMergeSubsegments(subsegmentation.pre, subsegmentation.post, subsegmentation.union)) {
+      if(!finalize && PathSegmenter.shouldLinkSubsegments(fullMergeSegmentation)) {
+        this.lingeringSubsegmentations.push(subsegmentation);
+      } else {
+        // Merge all as a completed segment!
+        this._protoSegments.push(predecessor);
+        this._protoSegmentSets.push([...this.lingeringSubsegmentations.map((val) => val.pre), subsegmentation.pre]);
+        this.lingeringSubsegmentations = [];
+
+        console.log("Proto-segments length: " + this._protoSegments.length);
+      }
+    }
+
+    /**
+     * Used by the 'filter' step (phase 2 of segmentation) to determine whether or not
+     * two subsegments should be 'linked' as members of the same segment.  (That is, if
+     * a user would likely consider the two subsegments as belonging to the "same arc
+     * of motion".)
+     * @param segmentation
+     * @returns
+     */
+    private static shouldLinkSubsegments(segmentation: Segmentation): boolean {
+      // // Known case #1:
+      // //   Near-identical direction, but heavy speed difference.
+      // //   Speed's still high enough to not be a 'wait'.
+      // //   We may want to 'note' the higher speed; that may be relevant for flick detection.
+      // // Known case #2:
+      // //   Low-speed pivot; angle change caught on very low velocity for both subsegments.
+      // //   ... or should this be merged?  Multiple mini-segments from 'wiggling' could just go ignored instead...
+
+      // This is kinda plain and arbitrary, but it seems to work "well enough" for now,
+      // at least at this stage of development.  (Most testing was done with Chrome emulation of
+      // an iPhone SE.)
+
+      // .mean('v') < 80:  the mean speed (taken timestamp to timestamp) does not exceed 80px/sec.
+      if(segmentation.pre.mean('v') < 80 && segmentation.post.mean('v') > 80) {
+        console.log("desegmentation exception");
+        return;
+      }
+      if(segmentation.pre.mean('v') > 80 && segmentation.post.mean('v') < 80) {
+        console.log("desegmentation exception");
+        return;
+      }
```

Comment:
This value of 80 pixels/sec seems to me to be likely to be dependent on the hardware. Higher-res devices will have a greater threshold, lower-res devices a lower threshold. So not sure we can depend on this.

In any case, this is a constant that should be defined somewhere outside this function.

Reply:
> This value of 80 pixels/sec seems to me to be likely to be dependent on the hardware. Higher-res devices will have a greater threshold, lower-res devices a lower threshold. So not sure we can depend on this.

I agree.  I settled for this for now because...

1. It at least gets us something functional that we can use to move forward.
2. There's probably some underlying principle that I can infer from how well it's working that should be adaptable to different resolutions.  It'd probably be helped by `devicePixelRatio` stuff, for one.

It's definitely a point worth revisiting and marking as a to-do on the base feature branch should a better solution be deferred for now.  (Which it probably will.)